stories/examples/zoom.stories.tsx

```diff
@@ -0,0 +1,279 @@
+import { Meta } from '@storybook/react';
+import { useRef, useState } from 'react';
+
+import { Axis, LineSeries, Plot, Annotations, ScatterSeries } from '../../src';
+import { DirectedEllipse } from '../../src/components/Annotations/DirectedEllipse';
+import { Rectangle } from '../../src/components/Annotations/Rectangle';
+import { DEFAULT_PLOT_CONFIG } from '../utils';
+
+export default {
+  title: 'Examples/Zoom',
+} as Meta;
+
+const data = [
+  { x: 1, y: 10 },
+  { x: 2, y: 5 },
+  { x: 3, y: 3 },
+  { x: 4, y: 5 },
+  { x: 5, y: 10 },
+  { x: 8, y: 5 },
+  { x: 9, y: 1 },
+];
+
+interface HorizontalPositions {
+  position?: {
+    x1: number;
+    x2: number;
+  } | null;
+  min?: number;
+  max?: number;
+}
+
+export function HorizontalZoom() {
+  const [{ position, min, max }, setPositions] =
+    useState<HorizontalPositions | null>({
+      position: null,
+      min: undefined,
+      max: undefined,
+    });
+  let click = useRef<boolean>(false);
+  return (
+    <div>
+      <Plot
+        {...DEFAULT_PLOT_CONFIG}
+        onMouseDown={({ coordinates: { x } }) => {
+          setPositions({
+            position: {
+              x1: x,
+              x2: x,
+            },
+            min: min,
+            max: max,
+          });
+          click.current = true;
+        }}
+        onMouseUp={() => {
+          click.current = false;
+          if (position.x1 !== position.x2) {
+            setPositions({
+              position: null,
+              min: Math.min(position.x1, position.x2),
+              max: Math.max(position.x1, position.x2),
+            });
+          }
+        }}
+        onMouseMove={({ coordinates: { x } }) => {
+          if (click.current) {
+            setPositions(({ position }) => ({
+              position: {
+                x1: position ? position.x1 : x,
+                x2: x,
+              },
+              min: min,
+              max: max,
+            }));
```

Comment:
It makes more sense to update all the values based on what is passed to the update function. Your min max values might not be what you expect here.

```suggestion
            setPositions((positions) => ({
              ...positions,
              position: {
                x1: positions.position ? positions.position.x1 : x,
                x2: x,
              },
            }));
```

Reply:
fixed